app/validators/capital_assessment/property_ownership_validator.rb

```diff
@@ -0,0 +1,54 @@
+module CapitalAssessment
+  class PropertyOwnershipValidator < ActiveModel::Validator
+    attr_reader :form
+
+    def validate(form)
+      @form = form
+
+      form.errors.add :percentage_applicant_owned, :invalid unless valid_ownership_total?(form)
+      form.errors.add :percentage_partner_owned, :invalid if client_has_partner(form) && !valid_ownership_total?(form)
+    end
+
+    private
+
+    def valid_ownership_total?(form)
+      return true if yet_to_add_other_owners?(form)
+      return true unless complete_property_owners?(form)
+
+      percentage_ownerships = all_percentage_ownerships(form)
+
+      percentage_ownerships.sum == 100
+    end
+
+    def all_percentage_ownerships(form)
+      percentage_ownerships = property_owners?(form) ? include_property_owners(form) : []
+      percentage_ownerships << form.percentage_applicant_owned unless form.percentage_applicant_owned.nil?
```

Comment:
Here I am checking if the percentage_applicant_owned is nil but technically the validation on the field to check it's presence would catch it so it shouldn't be nil at this stage

Reply:
No harm

Reply:
You could always run `.compact` on the array and that would remove any nil values that were added by percentage_applicant_owned.nil? or po.percentage_owned.nil?

Either way, not super necessary, but also no harm done your way